components/instructions/tools.tsx

```diff
@@ -122,3 +128,290 @@ export async function getInstructionDescriptor(
+//   const startTime = getUnixTs();
+
+//   const txid: any = await connection.sendRawTransaction(
+//     rawTransaction,
+//     {
+//       skipPreflight: true,
+//     },
+//   );
+
+//   console.log('Started awaiting confirmation for', txid);
+
+//   let done = false;
+//   (async () => {
+//     while (!done && getUnixTs() - startTime < timeout) {
+//       connection.sendRawTransaction(rawTransaction, {
+//         skipPreflight: true,
+//       });
+//       await sleep(3000);
+//     }
+//   })();
+//   try {
+//     await awaitTransactionSignatureConfirmation(txid, timeout, connection);
+//   } catch (err) {
+//     if ((err as any).timeout) {
+//       throw new Error(txid);
+//     }
+//     let simulateResult: any | null = null;
+//     try {
+//       simulateResult = (
+//         await simulateTransaction(connection, signedTransaction, 'single')
+//       ).value;
+//     } catch (e) {
+//       console.log('Error: ', e);
+//     }
+
+//     if (simulateResult && simulateResult.err) {
+//       if (simulateResult.logs) {
+//         for (let i = simulateResult.logs.length - 1; i >= 0; --i) {
+//           const line = simulateResult.logs[i];
+//           if (line.startsWith('Program log: ')) {
+//             throw new Error(
+//               simulateResult.err,
+//             );
+//           }
+//         }
+//       }
+//       throw new Error(
+//         simulateResult.err,
+//       );
+//     }
+
+//     throw new Error('Transaction failed');
+//   } finally {
+//     done = true;
+//   }
+
+//   console.log('Latency', txid, getUnixTs() - startTime);
+//   return txid;
+// }
+
+// export async function sendTransaction2({
+//   transaction,
+//   wallet,
+//   signers = [],
+//   connection,
+//   timeout = 3000,
+// }: {
+//   transaction: Transaction;
+//   wallet: any;
+//   signers?: Array<Account>;
+//   connection: Connection;
+//   sendingMessage?: string;
+//   successMessage?: string;
+//   timeout?: number;
+// }) {
+//   const signedTransaction = await signTransaction({
+//     transaction,
+//     wallet,
+//     signers,
+//     connection,
+//   });
+//   return await sendSignedTransaction({
+//     signedTransaction,
+//     connection,
+//     timeout,
+//   });
+// }
+
+// export async function sendTransactionWithNotifications(
+//   connection: Connection,
+//   wallet: any,
+//   instructions: TransactionInstruction[],
+//   signers: Account[],
+//   pendingMessage: string,
+//   successMessage: string,
+// ) {
+//   try {
+//     const transaction = new Transaction();
+//     transaction.add(...instructions);
+
+//     try {
+//       let txid = await sendTransaction2({
+//         transaction,
+//         wallet,
+//         signers,
+//         connection,
+//       });
+
+//       console.log('siccess', txid)
+//     } catch (txError) {
+//       throw new Error(txError);
+//     }
+//   } catch (ex) {
+//     console.error(ex);
+//     throw ex;
+//   }
+// }
```

Comment:
Please, leave a reason for this decision, and also, be sure that there are no side effects by doing this